File: packages/dapp/js/lib/realitycheck-template.js
'use strict';

const TEMPLATE_CONFIG = require('../config/templates.json');

exports.defaultTemplateIDForType = function(template_type) {
  return TEMPLATE_CONFIG.base_ids[template_type];
}

exports.defaultTemplateForType = function(template_type) {
  return TEMPLATE_CONFIG.content[this.defaultTemplateIDForType(template_type)];
}

exports.preloadedTemplateContents = function() {
  return TEMPLATE_CONFIG.content;
}

exports.encodeText = function(qtype, txt, outcomes, category) {
	var qtext = JSON.stringify(txt).replace(/^"|"$/g, '');
	//console.log('using template_id', template_id);
	if (qtype == 'single-select' || qtype == 'multiple-select') {
	var outcome_str = JSON.stringify(outcomes).replace(/^\[/, '').replace(/\]$/, '');
	//console.log('made outcome_str', outcome_str);
	qtext = qtext + QUESTION_DELIMITER + outcome_str;
	//console.log('made qtext', qtext);
	}
    qtext = qtext + QUESTION_DELIMITER + category;
   return qtext;
}

File: packages/python-services/src/rcpy/rcindex/realitycheck_template.py
import json

class RealityCheckTemplate:

    @staticmethod
    def templateConfig():
        with open('rcindex/config/templates.json') as f:
            return json.load(f)

    @staticmethod
    def defaultTemplateIDForType(template_type):
        config = RealityCheckTemplate.templateConfig()
        return config['base_ids'][template_type]

    @staticmethod
    def defaultTemplateForType(template_type):
        content = RealityCheckTemplate.preloadedTemplateContents()
        # print(content)
        template_id = str(RealityCheckTemplate.defaultTemplateIDForType(template_type))
        # print(template_id)
        return content[template_id]

    @staticmethod
    def preloadedTemplateContents():
        config = RealityCheckTemplate.templateConfig()
        return config['content']
